### PR TITLE
Prepare 1.5.1

### PR DESCRIPTION
**Pull request by jneira**
* The hackage check is [here](https://github.com/jneira/haskell-language-server/runs/4305202883?check_suite_focus=true). It is failing for ghc-8.6.5 with

```
cabal: Could not resolve dependencies:
[__0] trying: ghcide-1.5.0.1 (user goal)
[__1] rejecting: ghcide:!test (constraint from config file, command line flag,
or user target requires opposite flag selection)
[__1] trying: ghcide:*test
[__2] trying: lsp-test-0.14.0.1 (dependency of ghcide *test)
[__3] trying: conduit-parse-0.2.1.0 (dependency of lsp-test)
[__4] trying: parsers-0.12.10 (dependency of conduit-parse)
[__5] trying: semigroups-0.20 (dependency of parsers)
[__6] trying: text-1.2.3.1/installed-1.2.3.1 (dependency of ghcide)
[__7] next goal: sqlite-simple (dependency of ghcide)
[__7] rejecting: sqlite-simple-0.4.18.0 (conflict: semigroups==0.20,
sqlite-simple => semigroups>=0.18 && <0.20)
[__7] skipping: sqlite-simple-0.4.16.0, sqlite-simple-0.4.15.0 (has the same
characteristics that caused the previous version to fail: excludes
'semigroups' version 0.20)
[__7] rejecting: sqlite-simple-0.4.14.0, sqlite-simple-0.4.13.0,
sqlite-simple-0.4.12.1, sqlite-simple-0.4.12.0, sqlite-simple-0.4.11.0,
sqlite-simple-0.4.10.0, sqlite-simple-0.4.9.0, sqlite-simple-0.4.8.0,
sqlite-simple-0.4.7.0, sqlite-simple-0.4.6.1, sqlite-simple-0.4.6.0,
sqlite-simple-0.4.5.2 (conflict: text => base==4.12.0.0/installed-4.12.0.0,
sqlite-simple => base>=4.5 && <4.11)
[__7] rejecting: sqlite-simple-0.4.5.1, sqlite-simple-0.4.5.0 (conflict: text
=> base==4.12.0.0/installed-4.12.0.0, sqlite-simple => base>=4.6 && <4.11)
[__7] rejecting: sqlite-simple-0.4.4.0, sqlite-simple-0.4.3.1,
sqlite-simple-0.4.3.0, sqlite-simple-0.4.2.0, sqlite-simple-0.4.1.0,
sqlite-simple-0.4.0.0, sqlite-simple-0.3.0.0 (conflict: text =>
base==4.12.0.0/installed-4.12.0.0, sqlite-simple => base<4.11)
[__7] trying: sqlite-simple-0.2.1.0
[__8] next goal: direct-sqlite (dependency of sqlite-simple)
[__8] rejecting: direct-sqlite-2.3.26 (conflict: sqlite-simple =>
direct-sqlite>=2.2 && <2.3)
[__8] skipping: direct-sqlite-2.3.24, direct-sqlite-2.3.23,
direct-sqlite-2.3.22, direct-sqlite-2.3.21, direct-sqlite-2.3.20,
direct-sqlite-2.3.19, direct-sqlite-2.3.18, direct-sqlite-2.3.17,
direct-sqlite-2.3.16, direct-sqlite-2.3.15, direct-sqlite-2.3.14,
direct-sqlite-2.3.13, direct-sqlite-2.3.12, direct-sqlite-2.3.11,
direct-sqlite-2.3.9, direct-sqlite-2.3.8, direct-sqlite-2.3.7,
direct-sqlite-2.3.6, direct-sqlite-2.3.5, direct-sqlite-2.3.4,
direct-sqlite-2.3.3.1, direct-sqlite-2.3.3, direct-sqlite-2.3.2,
direct-sqlite-2.3.1, direct-sqlite-2.3 (has the same characteristics that
caused the previous version to fail: excluded by constraint '>=2.2 && <2.3'
from 'sqlite-simple')
[__8] rejecting: direct-sqlite-2.2.1 (conflict:
text==1.2.3.1/installed-1.2.3.1, direct-sqlite => text>=0.11 && <1)
[__8] skipping: direct-sqlite-2.2, direct-sqlite-2.1, direct-sqlite-2.0 (has
the same characteristics that caused the previous version to fail: excludes
'text' version 1.2.3.1)
[__8] rejecting: direct-sqlite-1.1 (conflict: sqlite-simple =>
direct-sqlite>=2.2 && <2.3)
[__8] skipping: direct-sqlite-1.0 (has the same characteristics that caused
the previous version to fail: excluded by constraint '>=2.2 && <2.3' from
'sqlite-simple')
[__8] fail (backjumping, conflict set: direct-sqlite, sqlite-simple, text)
Backjump limit reached (currently 4000, change with --max-backjumps or try to
run with --reorder-goals).
```

But i cant reproduce it locally

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2393"></a>

### CODE REVIEW

**Comment by jneira** [comment]
* Most plugin version bumps are unnecessary but the actual hackage workflow is not able to make a diff which dont fail in front of revised .cabal files (which includes a `x-revision` field and some whitespace changes that diff dont like)
* Will fix it but i think this is time to do the release so i bumped them anyways

**Comment by jneira** [comment]
> The hackage check is here. It is failing for ghc-8.6.5 with

In my local builds cabal always picks semigroups-0.19.2 but in ci is picking 0.20.0 :facepalm:

**Comment by jneira** [comment]
Reproduced with `cabal-3.4.0.0 build ghcide -w ghc-8.6.5 --max-backjumps 4000 --enable-tests --enable-benchmarks`
It is succesfull with cabal-3.6.2.0

**Comment by jneira** [comment]
Reproduced with `cabal-3.4.0.0 build ghcide -w ghc-8.6.5 --max-backjumps 4000 --enable-tests --enable-benchmarks`
It is succesfull with cabal-3.6.2.0

**Review by pepeiborra — APPROVED** [review]

File: plugins/hls-eval-plugin/hls-eval-plugin.cabal [hunk 1]
```diff
@@ -65,7 +65,7 @@ library
     , ghc
     , ghc-boot-th
     , ghc-paths
-    , ghcide                >=1.5.0  && <1.6
+    , ghcide                >=1.5.0.1  && <1.6
```
Comment (pepeiborra) [inline plugins/hls-eval-plugin/hls-eval-plugin.cabal:68]: Oops, so then this wasn't a ghcide minor release, my bad. This is why we need http://github.com/kowainik/policeman !

**Review by pepeiborra — APPROVED** [review]

File: plugins/hls-eval-plugin/hls-eval-plugin.cabal (hunk 1 above)
Comment (jneira) [reply plugins/hls-eval-plugin/hls-eval-plugin.cabal:68]: Yeah, i revised the .cabal in hackage as well, see: #2396

**Review by fendor — APPROVED** [review]
LGTM, but I am not completely in the loop right now, so please take it with a grain of salt.

**Comment by jneira** [comment]
8th CI run 🤦‍♂️

**Review by Ailrun — APPROVED** [review]
LGTM too

**Review by michaelpj — APPROVED** [review]

**Comment by jneira** [comment]
The workflow is failing deterministically (4 consecutive fails) for hls-tactic-plugin, ubuntu and ghc-8.6.5:

```
2021-11-28T05:12:36.2681217Z Test suite tests: RUNNING...
2021-11-28T05:12:36.3147169Z tactics
2021-11-28T05:12:36.3149290Z   AutoTuple
2021-11-28T05:12:36.3150930Z     auto for tuple
2021-11-28T05:12:36.5376452Z       should always be able to discover an auto solution:                             OK (0.26s)
2021-11-28T05:12:36.5377742Z   CodeAction.Auto
2021-11-28T05:12:36.5378740Z     golden
2021-11-28T05:12:46.5515264Z       AutoSplitGADT (golden):                                                         FAIL (10.01s)
2021-11-28T05:12:46.5516687Z         uncaught exception: SessionException
2021-11-28T05:12:46.5518103Z         Timed out waiting to receive a message from the server.
2021-11-28T05:12:46.5520403Z         Use -p '/AutoSplitGADT (golden)/' to rerun this test only.
2021-11-28T05:12:56.5583047Z       GoldenEitherAuto (golden):                                                      FAIL (10.01s)
2021-11-28T05:12:56.5584425Z         uncaught exception: SessionException
2021-11-28T05:12:56.5585534Z         Timed out waiting to receive a message from the server.
2021-11-28T05:12:56.5587492Z         Use -p '/GoldenEitherAuto (golden)/' to rerun this test only.
2021-11-28T05:13:06.5662592Z       GoldenJoinCont (golden):                                                        FAIL (10.01s)
2021-11-28T05:13:06.5664328Z         uncaught exception: SessionException
2021-11-28T05:13:06.5665471Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:06.5667220Z         Use -p '/GoldenJoinCont (golden)/' to rerun this test only.
2021-11-28T05:13:16.5744373Z       GoldenIdentityFunctor (golden):                                                 FAIL (10.01s)
2021-11-28T05:13:16.5746004Z         uncaught exception: SessionException
2021-11-28T05:13:16.5751323Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:16.5775932Z         Use -p '/GoldenIdentityFunctor (golden)/' to rerun this test only.
2021-11-28T05:13:26.5822534Z       GoldenIdTypeFam (golden):                                                       FAIL (10.01s)
2021-11-28T05:13:26.5823953Z         uncaught exception: SessionException
2021-11-28T05:13:26.5825035Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:26.5828332Z         Use -p '/GoldenIdTypeFam (golden)/' to rerun this test only.
2021-11-28T05:13:36.5906500Z       GoldenEitherHomomorphic (golden):                                               FAIL (10.01s)
2021-11-28T05:13:36.5908627Z         uncaught exception: SessionException
2021-11-28T05:13:36.5909942Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:36.5919012Z         Use -p '/GoldenEitherHomomorphic (golden)/' to rerun this test only.
2021-11-28T05:13:46.6190895Z       GoldenNote (golden):                                                            FAIL (10.03s)
2021-11-28T05:13:46.6192413Z         uncaught exception: SessionException
2021-11-28T05:13:46.6193495Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:46.6195189Z         Use -p '/GoldenNote (golden)/' to rerun this test only.
2021-11-28T05:13:56.6463109Z       GoldenPureList (golden):                                                        FAIL (10.03s)
2021-11-28T05:13:56.6464567Z         uncaught exception: SessionException
2021-11-28T05:13:56.6465849Z         Timed out waiting to receive a message from the server.
2021-11-28T05:13:56.6467613Z         Use -p '/GoldenPureList (golden)/' to rerun this test only.
2021-11-28T05:14:06.6546855Z       GoldenListFmap (golden):                                                        FAIL (10.01s)
2021-11-28T05:14:06.6548504Z         uncaught exception: SessionException
2021-11-28T05:14:06.6549596Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:06.6551522Z         Use -p '/GoldenListFmap (golden)/' to rerun this test only.
2021-11-28T05:14:16.6714979Z       GoldenFromMaybe (golden):                                                       FAIL (10.01s)
2021-11-28T05:14:16.6716415Z         uncaught exception: SessionException
2021-11-28T05:14:16.6717500Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:16.6719327Z         Use -p '/GoldenFromMaybe (golden)/' to rerun this test only.
2021-11-28T05:14:26.6745837Z       GoldenFoldr (golden):                                                           FAIL (10.01s)
2021-11-28T05:14:26.6747226Z         uncaught exception: SessionException
2021-11-28T05:14:26.6748326Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:26.6776057Z         Use -p '/GoldenFoldr (golden)/' to rerun this test only.
2021-11-28T05:14:36.6863435Z       GoldenSwap (golden):                                                            FAIL (10.01s)
2021-11-28T05:14:36.6864764Z         uncaught exception: SessionException
2021-11-28T05:14:36.6865831Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:36.6896264Z         Use -p '/GoldenSwap (golden)/' to rerun this test only.
2021-11-28T05:14:46.6946130Z       GoldenFmapTree (golden):                                                        FAIL (10.01s)
2021-11-28T05:14:46.6947597Z         uncaught exception: SessionException
2021-11-28T05:14:46.6948685Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:46.6977537Z         Use -p '/GoldenFmapTree (golden)/' to rerun this test only.
2021-11-28T05:14:56.7046085Z       GoldenGADTAuto (golden):                                                        FAIL (10.01s)
2021-11-28T05:14:56.7047431Z         uncaught exception: SessionException
2021-11-28T05:14:56.7048522Z         Timed out waiting to receive a message from the server.
2021-11-28T05:14:56.7055983Z         Use -p '/GoldenGADTAuto (golden)/' to rerun this test only.
2021-11-28T05:15:06.7146193Z       GoldenSwapMany (golden):                                                        FAIL (10.01s)
2021-11-28T05:15:06.7147660Z         uncaught exception: SessionException
2021-11-28T05:15:06.7148753Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:06.7176305Z         Use -p '/GoldenSwapMany (golden)/' to rerun this test only.
2021-11-28T05:15:16.7283923Z       GoldenBigTuple (golden):                                                        FAIL (10.01s)
2021-11-28T05:15:16.7285376Z         uncaught exception: SessionException
2021-11-28T05:15:16.7296732Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:16.7299302Z         Use -p '/GoldenBigTuple (golden)/' to rerun this test only.
2021-11-28T05:15:26.7383691Z       GoldenShow (golden):                                                            FAIL (10.01s)
2021-11-28T05:15:26.7385009Z         uncaught exception: SessionException
2021-11-28T05:15:26.7385994Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:26.7387957Z         Use -p '/GoldenShow (golden)/' to rerun this test only.
2021-11-28T05:15:36.7486937Z       GoldenShowCompose (golden):                                                     FAIL (10.01s)
2021-11-28T05:15:36.7488811Z         uncaught exception: SessionException
2021-11-28T05:15:36.7489871Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:36.7506165Z         Use -p '/GoldenShowCompose (golden)/' to rerun this test only.
2021-11-28T05:15:46.7720728Z       GoldenShowMapChar (golden):                                                     FAIL (10.02s)
2021-11-28T05:15:46.7725651Z         uncaught exception: SessionException
2021-11-28T05:15:46.7726496Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:46.7827135Z         Use -p '/GoldenShowMapChar (golden)/' to rerun this test only.
2021-11-28T05:15:57.0390813Z       GoldenSuperclass (golden):                                                      FAIL (10.21s)
2021-11-28T05:15:57.0399358Z         uncaught exception: SessionException
2021-11-28T05:15:57.0400445Z         Timed out waiting to receive a message from the server.
2021-11-28T05:15:57.0497278Z         Use -p '/GoldenSuperclass (golden)/' to rerun this test only.
2021-11-28T05:16:07.1816312Z       GoldenSafeHead (golden):                                                        FAIL (10.11s)
2021-11-28T05:16:07.1819293Z         uncaught exception: SessionException
2021-11-28T05:16:07.1820366Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:07.1915798Z         Use -p '/GoldenSafeHead (golden)/' to rerun this test only.
2021-11-28T05:16:17.5569900Z       FmapBoth (golden):                                                              FAIL (10.36s)
2021-11-28T05:16:17.5576567Z         uncaught exception: SessionException
2021-11-28T05:16:17.5577441Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:17.5791485Z         Use -p '/FmapBoth (golden)/' to rerun this test only.
2021-11-28T05:16:27.6964239Z       RecordCon (golden):                                                             FAIL (10.19s)
2021-11-28T05:16:27.6966883Z         uncaught exception: SessionException
2021-11-28T05:16:27.6967864Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:27.7296309Z         Use -p '/RecordCon (golden)/' to rerun this test only.
2021-11-28T05:16:37.8448141Z       NewtypeRecord (golden):                                                         FAIL (10.14s)
2021-11-28T05:16:37.8450522Z         uncaught exception: SessionException
2021-11-28T05:16:37.9771835Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:38.0258234Z         Use -p '/NewtypeRecord (golden)/' to rerun this test only.
2021-11-28T05:16:48.2902096Z       FmapJoin (golden):                                                              FAIL (10.34s)
2021-11-28T05:16:48.2912746Z         uncaught exception: SessionException
2021-11-28T05:16:48.2913859Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:48.3028425Z         Use -p '/FmapJoin (golden)/' to rerun this test only.
2021-11-28T05:16:58.6420250Z       Fgmap (golden):                                                                 FAIL (10.42s)
2021-11-28T05:16:58.6423975Z         uncaught exception: SessionException
2021-11-28T05:16:58.6424870Z         Timed out waiting to receive a message from the server.
2021-11-28T05:16:58.6486454Z         Use -p '/Fgmap (golden)/' to rerun this test only.
2021-11-28T05:17:09.0725877Z       FmapJoinInLet (golden):                                                         FAIL (10.43s)
2021-11-28T05:17:09.0728380Z         uncaught exception: SessionException
2021-11-28T05:17:09.0729250Z         Timed out waiting to receive a message from the server.
2021-11-28T05:17:09.1383293Z         Use -p '/FmapJoinInLet (golden)/' to rerun this test only.
2021-11-28T05:17:19.4888442Z       AutoEndo (golden):                                                              FAIL (10.37s)
2021-11-28T05:17:19.4891860Z         uncaught exception: SessionException
2021-11-28T05:17:19.4892731Z         Timed out waiting to receive a message from the server.
2021-11-28T05:17:19.5231111Z         Use -p '/AutoEndo (golden)/' to rerun this test only.
2021-11-28T05:17:29.7719198Z       AutoEmptyString (golden):                                                       FAIL (10.31s)
2021-11-28T05:17:29.7724161Z         uncaught exception: SessionException
2021-11-28T05:17:29.7730481Z         Timed out waiting to receive a message from the server.
2021-11-28T05:17:29.7784206Z         Use -p '/AutoEmptyString (golden)/' to rerun this test only.
2021-11-28T05:17:40.2278265Z       AutoPatSynUse (golden):                                                         FAIL (10.44s)
2021-11-28T05:17:40.2284038Z         uncaught exception: SessionException
2021-11-28T05:17:40.2312989Z         Timed out waiting to receive a message from the server.
```

I am gonna ignore the test suite for taht combination to merge this asap and unblock the release and rest of pr

//cc @pepeiborra @isovector

**Comment by pepeiborra** [comment]
> The workflow is failing deterministically (4 consecutive fails) for hls-tactic-plugin, ubuntu and ghc-8.6.5:

> //cc @pepeiborra @isovector

Why do you think it's deterministic? It passes consistently for me in https://github.com/haskell/haskell-language-server/actions/runs/1510511159

**Comment by jneira** [comment]
> > The workflow is failing deterministically (4 consecutive fails) for hls-tactic-plugin, ubuntu and ghc-8.6.5:
> 
> > //cc @pepeiborra @isovector
> 
> Why do you think it's deterministic? It passes consistently for me in https://github.com/haskell/haskell-language-server/actions/runs/1510511159

Well they failed with this pr in 4 consecutive attempts after being succesful in previous steps.
To be correct it was determinitic enough to ignore them temporary :-P. The workflow was succesful after ignoring, maybe it was coincidence 🤷 
I am enabling them again in #2410 let's see how is going ci there

**Pull request merged**